test/factory/customSmartWalletFactory.test.ts

```diff
@@ -0,0 +1,537 @@
+          privateKey,
+          dataToSign
+        );
+
+        const otherAccount = Wallet.createRandom();
+
+        await expect(
+          customSmartWalletFactory.createUserSmartWallet(
+            otherAccount.address,
+            recoverer,
+            logicAddress,
+            index,
+            initParams,
+            signature
+          )
+        ).to.be.rejectedWith('Invalid signature');
+      });
+    });
+
+    describe('relayedUserSmartWalletCreation', function () {
+      let recoverer: string;
+      let index: number;
+      let smartWalletAddress: string;
+      let worker: SignerWithAddress;
+      let otherCaller: SignerWithAddress;
+      let token: UtilToken;
+      let logicAddress: string;
+      const initParams = '0x';
+
+      beforeEach(async function () {
+        recoverer = constants.AddressZero;
+        logicAddress = constants.AddressZero;
+        index = randomNumber();
+        smartWalletAddress =
+          await customSmartWalletFactory.getSmartWalletAddress(
+            owner.address,
+            recoverer,
+            logicAddress,
+            utils.keccak256(initParams),
+            index
+          );
+        [worker, otherCaller] = await ethers.getSigners();
+        ({ contract: token } = await deployContract<UtilToken, []>({
+          contractName: 'UtilToken',
+          constructorArgs: [],
+        }));
+        await token.mint(1000, smartWalletAddress);
+      });
+
+      it('should initialize the smart wallet in the expected address without paying fee', async function () {
+        const deployRequest = createDeployRequest(
+          {
+            from: owner.address,
+            tokenContract: token.address,
+            tokenAmount: '0',
+            tokenGas: '0',
+            recoverer: recoverer,
+            index: index,
+            relayHub: worker.address,
+          },
+          {
+            callForwarder: customSmartWalletFactory.address,
+          }
+        );
+
+        const { suffixData, signature } = signDeployRequest(
+          owner,
+          deployRequest,
+          customSmartWalletFactory.address,
+          chainId
+        );
+
+        const initialWorkerBalance = await token.balanceOf(worker.address);
+
+        await customSmartWalletFactory
+          .connect(worker)
+          .relayedUserSmartWalletCreation(
+            deployRequest.request,
+            suffixData,
+            worker.address,
+            signature
+          );
+
+        const smartWallet = await ethers.getContractAt(
+          'SmartWallet',
+          smartWalletAddress
+        );
+
+        const finalWorkerBalance = await token.balanceOf(worker.address);
+
+        expect(finalWorkerBalance).to.be.equal(initialWorkerBalance);
+        await expect(smartWallet.isInitialized()).to.eventually.be.true;
+      });
+
+      it('should initialize the smart wallet in the expected address paying fee', async function () {
+        const amountToPay = 500;
```

Comment:
I prefer using big values to be more realistic.

Reply:
updated